kanidmd/src/lib/tracing_tree/subscriber.rs

```diff
@@ -478,8 +478,20 @@ impl TreePreProcessed {
         let buf = &formatted_logs[..];
 
         match processed_logs.tree_io() {
-            TreeIo::Stdout => io::stdout().write_all(buf),
-            TreeIo::Stderr => io::stderr().write_all(buf),
+            TreeIo::Stdout => {
+                // BUG - we can't write to stdout/err directly because this breaks
+                // cargo test capturing of io.
+                // io::stdout().write_all(buf)
+                let s = unsafe { std::str::from_utf8_unchecked(buf) };
```

Comment:
We really should just make the `writer` in the `formatter.rs` be a `String`, so we can avoid this. This could go in a future PR though.

Reply:
Yes I had a look, but the issue is then serde_json with the buffer. :( So it would be a bigger pr I think

Reply:
Ah, didn't realize. There's a checked variant, [`str::from_utf8`](https://doc.rust-lang.org/stable/std/str/fn.from_utf8.html), that we should probably use instead of introducing `unsafe` more.